clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/CachedKafkaMetadataGetter.java

```diff
@@ -119,16 +113,21 @@ <T> T fetchMetadata(
           T newValue = valueSupplier.get();
           metadataCache.put(key, new ValueAndExpiryTime<>(newValue, System.nanoTime() + ttlNs));
         } catch (TopicDoesNotExistException ex) {
```

Comment:
I thought about it again, so when exception happens, shall we just reset the `cachedValue` to be null? so that the next lookup will retry, and this might simplify the logic.

Reply:
Hmm, yeah especially for before ready to serve this might be better as retry is more frequent. But maybe I should document more here as it changes existing behavior.

Reply:
Actually, I think this behavior is better.
Let us the Kafka keeps failing and we always use the staled data, which means suddenly, the future version will start switching soon, which will affect the online read.
And I don't think it will be very frequent since this function will use `computeIfAbsent`, so there will be at one in-flight request per key, right?